### ISSUE TITLE
ALE does not show errors and warnings for js/ts file on opening

### ISSUE THREAD

**Issue by pseudocc** [dense-analysis/ale#4383]
I am new to ALE, the lsp am using is "deno".
When I open a js/ts file, the warnings/errors does not show up.

## Information

**VIM version**

VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Sep 13 2022 09:35:02)
Included patches: 1-3995, 4563, 4646, 4774, 4895, 4899, 4901, 4919

Operating System: Ubuntu 22.04.1 LTS

## What went wrong

Like what I have mention in the beginning, warnings and errors does not show up on js/ts file opening,
while it works for py, sh, c, etc..

## Reproducing the bug

```bash
cat << EOF > bad.ts
let unused;
konsole.log('misspelled');
console.log('okay');
EOF
vi bad.ts
```

1. Open a js.ts file with errors/warnings.
2. Then this happened.
3. Then I do some nonsense modifications, erros/warnings show up.

### :ALEInfo
<details>
  <summary>Expand</summary>
<pre><code>ALEInfo
 Current Filetype: typescript
Available Linters: ['cspell', 'deno', 'eslint', 'standard', 'tslint', 'tsserver', 't
ypecheck', 'xo']
  Enabled Linters: ['cspell', 'deno', 'eslint', 'standard', 'tslint', 'tsserver', 't
ypecheck', 'xo']
  Ignored Linters: []
 Suggested Fixers:
  'deno' - Fix TypeScript using deno fmt.
  'dprint' - Pluggable and configurable code formatting platform
  'eslint' - Apply eslint --fix to a file.
  'prettier' - Apply prettier to a file.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every
line.
  'tslint' - Fix typescript files with tslint --fix.
line.
  'tslint' - Fix typescript files with tslint --fix.
  'xo' - Fix JavaScript/TypeScript files using xo --fix.
 Linter Variables:
let g:ale_typescript_standard_executable = 'standard'
let g:ale_typescript_standard_options = ''
let g:ale_typescript_standard_use_global = 0
let g:ale_typescript_tslint_config_path = ''
let g:ale_typescript_tslint_executable = 'tslint'
let g:ale_typescript_tslint_ignore_empty_files = 0
let g:ale_typescript_tslint_rules_dir = ''
let g:ale_typescript_tslint_use_global = 0
let g:ale_typescript_tsserver_config_path = ''
let g:ale_typescript_tsserver_executable = 'tsserver'
let g:ale_typescript_tsserver_use_global = 0
let g:ale_typescript_xo_executable = 'xo'
let g:ale_typescript_xo_options = ''
let g:ale_typescript_xo_use_global = 0
 Global Variables:
let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = v:null
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_disable_lsp = 0
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = v:null
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = v:null
let g:ale_list_window_size = v:null
let g:ale_loclist_msg_format = v:null
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = v:null
let g:ale_open_list = v:null
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_root = {}
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = v:null
let g:ale_sign_error = v:null
let g:ale_sign_info = v:null
let g:ale_sign_offset = v:null
let g:ale_sign_style_error = v:null
let g:ale_sign_style_warning = v:null
let g:ale_sign_warning = v:null
let g:ale_sign_highlight_linenrs = v:null
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - failure) cspell
(executable check - success) deno
(executable check - failure) eslint
(executable check - failure) standard
(executable check - failure) tslint
(executable check - failure) tsserver
(executable check - failure) typecheck
(executable check - failure) xo
(started) ['/bin/bash', '-c', '''deno'' lsp']
<code/>
</details>

**Comment by pseudocc** [comment]
```
==== start log session ====
  0.006470 : raw terminal output: "[?1049h[22;0;0t[>4;2m[?1h=[?2004h[?1004h"
  0.006492 : raw terminal output: "[1;47r[?12h[?12l[22;2t"
  0.009819 : raw key input: "[I"
  0.074100 : Starting job: /bin/bash  -c  'deno' lsp
  0.074119 on 0: Created channel
  0.074736 : closing logfile /tmp/chlogfile.log
  0.075063 SEND on 0(in): 'Content-Length: 1335

{"method":"initialize","jsonrpc":"2.0","id":1,"params":{"initializationOptions":{"unstable":false,"enable":true,"lint":true,"importMap":""},"rootUri":"file:///home/catlas/tmp","capabilities":{"workspace":{"workspaceFolders":false,"configuration":false,"symbol":{"dynamicRegistration":false},"applyEdit":false,"didChangeConfiguration":{"dynamicRegistration":false}},"textDocument":{"documentSymbol":{"dynamicRegistration":false,"hierarchicalDocumentSymbolSupport":false},"references":{"dynamicRegistration":false},"publishDiagnostics":{"relatedInformation":true},"rename":{"dynamicRegistration":false},"completion":{"completionItem":{"snippetSupport":false,"commitCharactersSupport":false,"preselectSupport":false,"deprecatedSupport":false,"documentationFormat":["plaintext"]},"contextSupport":false,"dynamicRegistration":false},"synchronization":{"didSave":true,"willSaveWaitUntil":false,"willSave":false,"dynamicRegistration":false},"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":[]}},"dynamicRegistration":false},"typeDefinition":{"dynamicRegistration":false},"hover":{"dynamicRegistration":false,"contentFormat":["plaintext"]},"implementation":{"dynamicRegistration":false,"linkSupport":false},"definition":{"dynamicRegistration":false,"linkSupport":false}}},"rootPath":"/home/catlas/tmp","processId":51594}}'
  0.077676 : raw terminal output: "[2;1H▽[6n"
  0.077685 : raw terminal output: "[2;1H  [3;1HPzz\[0%m[6n"
  0.077707 : raw terminal output: "[>c"
  0.077713 : raw terminal output: "]10;?]11;?"
  0.078142 : SafeState: Start triggering
  0.079362 : raw key input: "[2;2R[3;1R[>0;276;0c]10;rgb:baba/baba/baba\]11;rgb:0000/0000/0000\"
  0.080567 : looking for messages on channels
  0.080578 : SafeState: back to waiting, triggering SafeStateAgain
  0.080604 : raw terminal output: "P+q436f\P+q6b75\P+q6b64\P+q6b72\P+q6b6c\P+q2332\P+q2334\P+q2569\P+q2a37\P+q6b31\"
  0.080620 : raw terminal output: "[?12$p"
  0.080633 : looking for messages on channels
  0.080636 : SafeState: back to waiting, triggering SafeStateAgain
  0.081723 RECV on 0(err): 'Starting Deno language server...
'
  0.081726 on 0: Invoking channel callback <lambda>5
  0.081738 : SafeState: back to waiting, triggering SafeStateAgain
  0.081747 RECV on 0(err): '  version: 1.28.3 (release, x86_64-unknown-linux-gnu)
'
  0.081748 : looking for messages on channels
  0.081750 on 0: Invoking channel callback <lambda>5
  0.081754 : SafeState: back to waiting, triggering SafeStateAgain
  0.081759 RECV on 0(err): '  executable: /home/catlas/.cargo/bin/deno
'
  0.081760 : looking for messages on channels
  0.081761 on 0: Invoking channel callback <lambda>5
  0.081765 : SafeState: back to waiting, triggering SafeStateAgain
  0.090916 : raw key input: "[?12;2$y"
  0.090971 : looking for messages on channels
  0.090978 : SafeState: back to waiting, triggering SafeStateAgain
  0.156307 RECV on 0(out): 'Content-Length: 1717

{"jsonrpc":"2.0","result":{"capabilities":{"textDocumentSync":{"openClose":true,"change":2,"save":{}},"selectionRangeProvider":true,"hoverProvider":true,"completionProvider":{"resolveProvider":true,"triggerCharacters":[".","\"","'","`","/","@","<","#"],"allCommitCharacters":[".",",",";","("]},"signatureHelpProvider":{"triggerCharacters":[",","(","<"],"retriggerCharacters":[")"]},"definitionProvider":true,"typeDefinitionProvider":true,"implementationProvider":true,"referencesProvider":true,"documentHighlightProvider":true,"documentSymbolProvider":{"label":"Deno"},"workspaceSymbolProvider":true,"codeActionProvider":{"codeActionKinds":["quickfix","refactor","refactor.extract.function","refactor.extract.constant","refactor.extract.type","refactor.extract.interface","refactor.move.newFile","refactor.rewrite.import","refactor.rewrite.export","refactor.rewrite.arrow.braces","refactor.rewrite.parameters.toDestructured","refactor.rewrite.property.generateAccessors"],"resolveProvider":true},"codeLensProvider":{"resolveProvider":true},"documentFormattingProvider":true,"renameProvider":true,"foldingRangeProvider":true,"workspace":{"workspaceFolders":{"supported":true,"changeNotifications":true}},"callHierarchyProvider":true,"semanticTokensProvider":{"legend":{"tokenTypes":["class","enum","interface","namespace","typeParameter","type","parameter","variable","enumMember","property","function","method"],"tokenModifiers":["declaration","static","async","readonly","defaultLibrary","local"]},"range":true,"full":true},"inlayHintProvider":true,"experimental":{"denoConfigTasks":true,"testingApi":true}},"serverInfo":{"name":"deno-language-server","version":"1.28.3 (release, x86_64-unknown-linux-gnu)"}},"id":1}'
  0.156316 on 0: Invoking channel callback <SNR>76_VimOutputCallback
  0.156646 SEND on 0(in): 'Content-Length: 52

{"method":"initialized","jsonrpc":"2.0","params":{}}'
  0.156750 RECV on 0(err): 'Server ready.
'
  0.156857 SEND on 0(in): 'Content-Length: 227

{"method":"textDocument/didOpen","jsonrpc":"2.0","params":{"textDocument":{"uri":"file:///home/catlas/tmp/bad.ts","version":1,"languageId":"typescript","text":"let unused;\nkonsole.log('misspelled');\nconsole.log('okay');\n"}}}'
  0.157051 on 0: Invoking channel callback <lambda>5
  0.157067 : SafeState: back to waiting, triggering SafeStateAgain
  0.157319 RECV on 0(out): 'Content-Length: 143

{"jsonrpc":"2.0","method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/catlas/tmp/bad.ts","section":"deno"}]},"id":0}'
  0.157323 : looking for messages on channels
  0.157337 on 0: Invoking channel callback <SNR>76_VimOutputCallback
  0.157612 : SafeState: back to waiting, triggering SafeStateAgain
  3.771787 : raw key input: ":"
  3.771860 : SafeState: reset: key typed
  3.775824 : SafeState: Start triggering
  3.775881 : looking for messages on channels
  3.775893 : SafeState: back to waiting, triggering SafeStateAgain
  4.777949 : raw key input: "q"
  4.778025 : SafeState: reset: key typed
  4.778050 : SafeState: Start triggering
  4.778082 : looking for messages on channels
  4.778094 : SafeState: back to waiting, triggering SafeStateAgain
  5.937691 : raw key input: "
"
  5.937769 : SafeState: reset: key typed
  5.937988 : Exiting...
  5.939631 : raw terminal output: "[?25l[?2004l[>4;m"
  5.939689 : raw terminal output: "[47;1H[K[47;1H[?1004l[?2004l[?1l>"
```

**Comment by w0rp** [comment]
This issue is likely stale by this point. The problems here were likely due to character encoding issues.
